### PR TITLE
Fix CreateRun response, output URI, and GetActionData

### PR DESCRIPTION
**Pull request by pingsutw**
## Tracking issue

Related to #6991

## Why are the changes needed?

1. **CreateRunResponse was incomplete** — only returned the action ID without metadata or status from the DB model
2. **Output URI was missing `/outputs.pb` suffix** — the executor's `outputRefs` stored the prefix path (e.g., `s3://flyte-data/<id>`) instead of the full file path (`s3://flyte-data/<id>/outputs.pb`), causing `GetActionData` to fail reading outputs
3. **GetActionData was fragile** — constructed output paths manually from `runOutputBase` instead of reading the URI stored in action attempts

## What changes were proposed in this pull request?

- **`runs/service/run_service.go`**: Use `convertRunToProto(run)` for `CreateRunResponse`; refactor `GetActionData` to read outputs from attempt URIs with concurrent `errgroup`; set root action name to `runId.Name`
- **`executor/pkg/controller/taskaction_controller.go`**: Append `/outputs.pb` to output URI in `outputRefs`
- **`runs/repository/impl/action.go`**: Append `/inputs.pb` to inputUri when creating run; set initial `Attempts` to 1; remove debug logging
- **`runs/repository/models/action.go`**: Add `Attempts` field to Action model
- **`runs/repository/interfaces/action.go`**: Rename `inputUri` param to `inputPrefix` for clarity

## How was this patch tested?

Unit tests added:

- `TestFillDefaultInputs` — default input filling with various edge cases
- `TestTaskIdFromTaskSpec` — task ID extraction from spec
- `TestGenerateCacheKeyForTask` — cache key generation determinism and input filtering
- `TestCreateRun_ResponseUsesRunModel` — verifies response populated from DB model
- `TestCreateRun_ActionIDUsesRunName` — verifies action ID uses run name
- `TestGetActionData_ReadsOutputFromAttempts` — verifies output read from attempt URI
- `TestGetActionData_NonSucceededSkipsOutputs` — verifies output skipped for non-terminal actions
- `TestOutputRefs` — verifies executor output URI includes `/outputs.pb`

### Labels

- **fixed**: Bug fix for output URI and CreateRun response
- **changed**: GetActionData refactored to use attempt URIs

## Check all the applicable boxes

- `main` 
  - \#6583
    - **Fix CreateRun response, output URI, and GetActionData** :point\_left:

### CODE REVIEW

File: runs/repository/impl/action_test.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ func setupActionDB(t *testing.T) *gorm.DB {
 		updated_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
 		ended_at DATETIME,
 		duration_ms INTEGER,
+		attempts INTEGER NOT NULL DEFAULT 0,
```
Comment (machichima) [inline runs/repository/impl/action_test.go]: Should we start from 1 or 0?

Comment (pingsutw) [reply runs/repository/impl/action_test.go]: 1

Comment (pingsutw) [reply runs/repository/impl/action_test.go]: updated

File: runs/service/run_service.go [hunk 2]
```diff
@@ -125,32 +128,58 @@ func (s *RunService) CreateRun(
...
-		org = id.RunId.Org
-		project = id.RunId.Project
-		domain = id.RunId.Domain
-		name = id.RunId.Name
+		runId = request.GetRunId()
 	case *workflow.CreateRunRequest_ProjectId:
-		org = id.ProjectId.Organization
-		project = id.ProjectId.Name
-		domain = id.ProjectId.Domain
-		name = generateRunName(time.Now().UnixNano())
```
Comment (machichima) [inline runs/service/run_service.go:138]: Seems like we didn't set value for those variables, we will get empty values below

File: runs/service/run_service.go
Comment (machichima) [inline runs/service/run_service.go]: We should use `inputs` here rather than `req.Msg.Inputs`?

File: runs/service/run_service.go [hunk 3]
```diff
@@ -162,66 +191,47 @@ func (s *RunService) CreateRun(
...
+			RunOutputBase: runOutputBase,
+			Spec: &actions.Action_Task{
+				Task: &workflow.TaskAction{
+					Id:       taskID,
+					Spec:     taskSpec,
+					CacheKey: wrapperspb.String(cacheKey),
+				},
+			},
+		},
+		RunSpec: request.RunSpec,
```
Comment (machichima) [inline runs/service/run_service.go]: I think we should use `runSpec` here rather than `request.RunSpec`?

File: runs/service/run_service.go [hunk 4]
```diff
@@ -576,55 +587,100 @@ func (s *RunService) GetActionData(
...
+			}
+
+			if len(attempts) == 0 {
+				return app.NewServerError(codes.NotFound, "outputs not available, no attempts for action")
+			}
+
+			outputUri := attempts[len(attempts)-1].GetOutputs().GetOutputUri()
+			if outputUri == "" {
+				return app.NewServerError(codes.NotFound, "outputs not available")
+			}
```
Comment (machichima) [inline runs/service/run_service.go:662]: Should we fallback to check if the output is in object store rather than directly fail here? Will there be a case that the output is already being written into the object store but DB hasn't updated

File: runs/service/run_service.go (hunk 2 above)
Comment (pingsutw) [reply runs/service/run_service.go:138]: good catch, thanks. I remove those variables and use runId.GetOrg(), runId.GetProject() instead.

File: runs/service/run_service.go (hunk 3 above)
Comment (pingsutw) [reply runs/service/run_service.go]: fixed

File: runs/service/run_service.go (hunk 4 above)
Comment (pingsutw) [reply runs/service/run_service.go:662]: No, executor always write the output first before sending the event (output uri is in the event).

File: runs/service/run_service.go
Comment (pingsutw) [reply runs/service/run_service.go]: fixed
